lib/upload.js

```diff
@@ -1014,6 +1097,96 @@ function splitSizeIntoParts (totalSize, partCount) {
   return parts
 }
 
+/**
+ * Build the chunk staggers.
+ *
+ * NOTE: Must be kept in sync with `splitSizeIntoParts` in skynet-js.
+ */
+function buildChunkStaggers (totalSize, partCount, chunkSize) {
```

Comment:
should we validate illegal parameters here? `partCount` being 0 etc, it makes it so we can safely execute the remainder of the function because there are some assumptions that part count is not zero for instance